satpy/readers/hdfeos_base.py

```diff
@@ -216,14 +217,48 @@ def load_dataset(self, dataset_name, is_category=False):
         from satpy.readers.hdf4_utils import from_sds
 
         dataset = self._read_dataset_in_file(dataset_name)
-        dask_arr = from_sds(dataset, chunks=CHUNK_SIZE)
+        chunks = self._chunks_for_variable(dataset)
+        dask_arr = from_sds(dataset, chunks=chunks)
         dims = ('y', 'x') if dask_arr.ndim == 2 else None
         data = xr.DataArray(dask_arr, dims=dims,
                             attrs=dataset.attributes())
         data = self._scale_and_mask_data_array(data, is_category=is_category)
 
         return data
 
+    def _chunks_for_variable(self, hdf_dataset):
+        scan_length_250m = 40
+        var_shape = hdf_dataset.info()[2]
+        res_multiplier = self._get_res_multiplier(var_shape)
+        non_yx_chunks = tuple()
+        if len(var_shape) == 3:
+            # assume (band, y, x)
```

Comment:
Is this documented somewhere? If it is, this comment should be rephrased to remove "assume"

Reply:
Depends what you mean by "documented". I suppose there could be some MODIS L2 files out there somewhere that have a 3D variable that doesn't have a "band" dimension. Due to some of the magic of the MODIS readers reading additional variables it wasn't easy to use other logic from things like `get_dataset` where we know the dimensions.

Other than the odd L2 product that I'm not aware of, I think all L1b variables are 2D or 3D...wait:

```
        byte DC\ Restore\ Change\ for\ Thermal\ Bands(number\ of\ scans, number\ of\ emissive\ bands, detectors\ per\ 1km\ band) ;
        byte DC\ Restore\ Change\ for\ Reflective\ 250m\ Bands(number\ of\ scans, number\ of\ 250m\ bands, detectors\ per\ 250m\ band) ;
        byte DC\ Restore\ Change\ for\ Reflective\ 500m\ Bands(number\ of\ scans, number\ of\ 500m\ bands, detectors\ per\ 500m\ band) ;
        byte DC\ Restore\ Change\ for\ Reflective\ 1km\ Bands(number\ of\ scans, number\ of\ 1km\ reflective\ bands, detectors\ per\ 1km\ band) ;
```

These are in a IMAPP L1B file I have. Not sure the right approach here.

Reply:
can we extract the dimension names from the hdf file?

Reply:
If pyhdf puts the dimensions on the variable object then I suppose we could, but since this code is used for L1b and L2 I'm not sure there is a good/stable way of doing dimension name -> meaning.